app/src/main/java/com/hedvig/app/feature/embark/passages/previousinsurer/PreviousInsurerFragment.kt

```diff
@@ -0,0 +1,105 @@
+package com.hedvig.app.feature.embark.passages.previousinsurer
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.os.Parcelable
+import android.view.View
+import androidx.fragment.app.Fragment
+import com.hedvig.android.owldroid.graphql.EmbarkStoryQuery
+import com.hedvig.app.R
+import com.hedvig.app.databinding.FragmentPreviousInsurerBinding
+import com.hedvig.app.feature.embark.EmbarkViewModel
+import com.hedvig.app.feature.embark.passages.MessageAdapter
+import com.hedvig.app.feature.embark.ui.PreviousInsurerBottomSheet
+import com.hedvig.app.feature.embark.ui.PreviousInsurerBottomSheet.Companion.EXTRA_INSURER_ID
+import com.hedvig.app.util.extensions.view.setHapticClickListener
+import com.hedvig.app.util.extensions.viewBinding
+import kotlinx.android.parcel.Parcelize
+import org.koin.android.viewmodel.ext.android.sharedViewModel
+
+class PreviousInsurerFragment : Fragment(R.layout.fragment_previous_insurer) {
+
+    private val binding by viewBinding(FragmentPreviousInsurerBinding::bind)
+    private val model: EmbarkViewModel by sharedViewModel()
+
+    private val insurerData by lazy {
+        requireArguments()
+            .getParcelable<PreviousInsurerData>(DATA)
+            ?: throw IllegalArgumentException("No argument passed to ${this.javaClass.name}")
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        binding.apply {
+            messages.adapter = MessageAdapter(insurerData.messages)
+            currentInsurerContainer.setHapticClickListener {
+                onShowInsurers()
+            }
+        }
+    }
+
+    private fun onShowInsurers() {
+        val fragment = PreviousInsurerBottomSheet.newInstance(insurerData.previousInsurers)
+        fragment.setTargetFragment(this, REQUEST_SELECT_INSURER)
```

Comment:
We probably shouldn't use the `setTargetFragment`-way of communicating between these fragments, given its imminent deprecation 😧

Reply:
I think using a shared view model here is a bit overkill since we just want to return a string when we are done with the bottom sheet fragment. A view model makes more sense if we wanted to share the data and keep some form of state between the fragments. 

The reason why this is going to be deprecated is because we cant use different fragment managers atm, but thats not a problem in this case. 

I suggest we keep this here and replace with the new API when its released!

Reply:
I'd actually prefer if we upgrade to the 1.3.0 RC and use the new API, given that we're not releasing this code for at least a few weeks. What do you think about that? 🙂

Reply:
Sounds good!